### PR TITLE
Adding support to compile to WebAssembly

### PR DESCRIPTION
**Pull request by shahrodkh**
I have made additions to your Makefile to compile to a .wasm file instead of .asm
Some other minor changes had to be made so that it will compile.

### CODE REVIEW

**Comment by fta2012** [comment]
Cool I didn't know web assembly was ready!

Some comments:

- It's my fault that the make file isn't organized better but there should multiple sets of flags and build targets. They are not named but the existing grouping of flags are roughly `mandatory`, `release`, and `debug` (where the debug flags are currently commented out and aren't tested/used).
  - I am mentioning this because I recall docs saying that the flags you enabled (`ASSERTIONS`, `ALLOW_MEMORY_GROWTH`) will prevent certain optimizations so probably shouldn't be in the release build.
  - Likewise the `-s WASM` flag should probably be a target specific variable for a new wasm build target . Then you can have both the wasm and regular build and choose which to use depending on browser support
- What is the sed replacement for?  I think there are callbacks like Module.preInit/preRun/postRun that you can hook into more cleanly if that's what you're going for. If that fails you can also add `EM_ASM` to the `int main()` in bindings to make it call your callback. But testing it out myself it seems like postRun will work.
- I don't like adding a completely new demo since you can't compare it with the non-wasm build. Also a quicker way to do a static webserver is `sudo python -m SimpleHTTPServer 80` instead of adding a boilerplate express server.

I am glad you got the wasm working but overall I don't really like this change. If you can simplify it to just adding a new build target and adding wasm feature detection for the existing demo, then that would be okay with me!

I made some quick changes in https://github.com/fta2012/ccv-js/commit/f63c56c81f38e15e41d799485b820bfa3656abb9 to reflect some of the issues I mentioned above. If you're still interested in doing so you can rebase and try adding the wasm target again.

Thanks!

File: ccv_bindings.cpp [hunk 1]
```diff
@@ -242,10 +242,10 @@ val ccv_dense_matrix_t_get_data(const std::shared_ptr<ccv_dense_matrix_t>& point
       return val(typed_memory_view(numElement, pointer->data.i32));
     case CCV_32F:
       return val(typed_memory_view(numElement, pointer->data.f32));
-    case CCV_64S:
-      return val(typed_memory_view(numElement, pointer->data.i64));
-    case CCV_64F:
-      return val(typed_memory_view(numElement, pointer->data.f64));
+    // case CCV_64S:
+    //   return val(typed_memory_view(numElement, pointer->data.i64));
+    // case CCV_64F:
```
Comment (fta2012) [inline ccv_bindings.cpp:247]: I don't think this case also needs to be commented out. There are no Uint64Array in js but there are Float64Array.

**Comment by fta2012** [comment]
Btw I changed it so that it will look for a global CCV variable to call CCVLib with (checkout index.html). I haven't tried this but you should be able to add your wasmBinary there too before loading the script:
 
```
// load your wasm
...
CCV = {
  wasmBinary: buffer,
  postRun: [ whatever ]
};
...
// load ccv.js 
```

**Comment by shahrodkh** [comment]
hey got caught up elsewhere -- looks like you made the changes though? hopefully I'll get a chance to check it out this weekend 👍

**Pull request closed without merging**